src/Databases/AlterDatabaseCommands.cpp

```diff
@@ -0,0 +1,413 @@
+    {
+        switch (type)
+        {
+            case ASTAlterCommand::ADD_TABLE_OVERRIDE:
+                return "ADD";
+            case ASTAlterCommand::MODIFY_TABLE_OVERRIDE:
+                return "MODIFY";
+            case ASTAlterCommand::DROP_TABLE_OVERRIDE:
+                return "DROP";
+            default:
+                throw Exception(ErrorCodes::LOGICAL_ERROR, "invalid ASTAlterCommand::Type {} - this is a bug!", ASTAlterCommand::typeToString(type));
+        }
+    }
+}
+
+void AlterDatabaseTableOverrideCommand::prepare(const DatabasePtr & database, ContextPtr context, AlterDatabaseApplyData & apply_data)
+{
+    if (apply_data.overrides_already_seen_in_query.contains(table_override->table_name))
+    {
+        throw Exception(
+            ErrorCodes::INVALID_ALTER_DATABASE_QUERY,
+            "Multiple ALTER DATABASE {{ADD|MODIFY|DROP}} TABLE OVERRIDE commands referring to override `{}`",
+            table_override->table_name);
+    }
+    apply_data.overrides_already_seen_in_query.insert(table_override->table_name);
+    ASTTableOverride * existing_override = tryGetExistingOverride(apply_data.getCreateQuery(), table_override->table_name);
+    if ((type == ASTAlterCommand::DROP_TABLE_OVERRIDE || type == ASTAlterCommand::MODIFY_TABLE_OVERRIDE) && !existing_override)
+    {
+        throw Exception(
+            ErrorCodes::TABLE_OVERRIDE_NOT_FOUND,
+            "ALTER DATABASE {} TABLE OVERRIDE refers to unknown override `{}`",
+            whichOverrideCommand(type),
+            table_override->table_name);
+    }
+    else if (type == ASTAlterCommand::ADD_TABLE_OVERRIDE && existing_override)
+    {
+        throw Exception(
+            ErrorCodes::TABLE_OVERRIDE_ALREADY_EXISTS,
+            "ALTER DATABASE {} TABLE OVERRIDE refers to existing override `{}`",
+            whichOverrideCommand(type),
+            table_override->table_name);
+    }
+
+    if (database->isTableExist(table_override->table_name, context))
+    {
+        if (table_override->modifiesStorage() != (existing_override && existing_override->modifiesStorage()))
+        {
+            /// Only the old or new override modifies storage
+            apply_data.need_to_recreate_database = true;
+            if (table_override->modifiesStorage())
+                apply_data.need_to_recreate_database_reason = "existing override does not modify storage, new override does";
+            else
+                apply_data.need_to_recreate_database_reason = "existing override modifies storage, new override does not";
+
+        }
+        else if (table_override->modifiesStorage())
+        {
+            /// Here, we know existing_override->storage exists after the previous if. If storage parameters
+            /// are different, the database needs to be recreated.
+            if (serializeAST(*table_override->storage) != serializeAST(*existing_override->storage))
+            {
+                apply_data.need_to_recreate_database = true;
```

Comment:
why need recreate database if only one table affected? why not just recreate table?

Reply:
Currently we have no way of recreating just a single table with MaterializedMySQL. I have another PR pending that will allow this.